### PR TITLE
consolidate autoPrune and autoPrunePath

### PR DESCRIPTION
**Pull request by krackerman**
fix: #673 consolidate autoPrune and autoPrunePath. Updated tests. removed autoPrunePath from fields-param

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -147,11 +147,11 @@ const Api = (options = {}) => {
     if (params.filter(p => p.position === 'path').some(p => request.indexOf(`{${p.nameOriginal}}`) === -1)) {
       throw new Error('Path Parameter not defined in given path.');
     }
-    if (params.filter(p => p.paramType === 'FieldsParam' && p.autoPrune === true).length > 1) {
+    if (params.filter(p => p.paramType === 'FieldsParam' && typeof p.autoPrune === 'string').length > 1) {
       throw new Error('Only one auto pruning "FieldsParam" per endpoint.');
     }
     endpoints[request] = params;
-    const rawAutoPruneFieldsParam = params.find(p => p.paramType === 'FieldsParam' && p.autoPrune === true);
+    const rawAutoPruneFieldsParam = params.find(p => p.paramType === 'FieldsParam');
```
Comment (MacMcIrish) [inline src/index.js]: Why'd this `p.autoPrune` get moved into the `pruneFields` function below?

File: test/response.spec.js [hunk 2]
```diff
@@ -234,7 +234,7 @@ describe('Testing Response', () => {
 
   it('Testing pruneResponse false', (done) => {
     api.wrap('GET test', [
-      api.FieldsParam('fields', 'query', { fields: ['foo'], autoPrune: false })
+      api.FieldsParam('fields', 'query', { fields: ['foo'], autoPrune: null })
```
Comment (MacMcIrish) [inline test/response.spec.js]: And then change the name of the test
```suggestion
      api.FieldsParam('fields', 'query', { fields: ['foo'] })
```

File: test/params/fields-param.spec.js [hunk 3]
```diff
@@ -39,8 +39,8 @@ describe('Testing FieldsParam Parameter', () => {
 
   it('Testing only one autoPrune FieldsParam per request', (done) => {
     expect(() => api.wrap('GET route', [
-      api.FieldsParam('fields1', 'query', { paths: ['id'], autoPrune: true }),
-      api.FieldsParam('fields2', 'query', { paths: ['id'], autoPrune: true })
+      api.FieldsParam('fields1', 'query', { paths: ['id'], autoPrune: 'payload' }),
+      api.FieldsParam('fields2', 'query', { paths: ['id'], autoPrune: 'payload' })
```
Comment (simlu) [inline test/params/fields-param.spec.js]:
```suggestion
      api.FieldsParam('fields2', 'query', { paths: ['id'], autoPrune: '' })
```

Comment (simlu) [reply test/params/fields-param.spec.js]: same above

File: test/response.spec.js [hunk 4]
```diff
@@ -180,7 +180,7 @@ describe('Testing Response', () => {
 
   it('Testing auto field pruning', (done) => {
```
Comment (simlu) [inline test/response.spec.js]:
```suggestion
  it('Testing auto field pruning top level', (done) => {
```

**Pull request merged**